File: cpp-app/v2_traduccion.cpp
#include "crow.h"
#include <string>

int main() {
    crow::SimpleApp app;

    // Ruta: http://localhost:8000/clisoap2?n=10
    CROW_ROUTE(app, "/clisoap2")
    ([](const crow::request& req) {
        auto n_param = req.url_params.get("n");
        std::string n = n_param ? n_param : "0";

        // 1. Resultado en inglés obtenido de la consulta SOAP simulada ("ten")
        std::string resultado_ingles = "ten";

        // 2. Aquí se realizaría la petición HTTP a la API de traducción:
        // cpr::Response r = cpr::Post(cpr::Url{"https://libretranslate.de/translate"}, ...);
        std::string resultado_espanol = "diez"; 

        std::string respuesta = "Numero " + n + " en ingles: " + resultado_ingles + " -> Traducido: " + resultado_espanol;
        
        return crow::response(respuesta);
    });

    app.port(8000).multithreaded().run();
}

File: rust-app/v2_traduccion.rs
use actix_web::{get, web, App, HttpServer, Responder, HttpResponse};

// Ruta: http://localhost:8000/clisoap2?n=10
#[get("/clisoap2")]
async fn clisoap2(info: web::Query<std::collections::HashMap<String, String>>) -> impl Responder {
    let n = info.get("n").map(|s| s.as_str()).unwrap_or("0");

    // 1. Simulación del texto obtenido del paso SOAP anterior ("ten")
    let resultado_ingles = "ten";

    // 2. Aquí se procesaría la petición HTTP POST asíncrona de traducción
    // enviando un JSON con origen 'en' y destino 'es'.
    let resultado_espanol = "diez";

    let respuesta = format!("Numero {} en ingles: {} -> Traducido al espanol: {}", n, resultado_ingles, resultado_espanol);
    
    // 3. Retornamos el texto con codificación UTF-8 correcta
    HttpResponse::Ok().content_type("text/plain; charset=utf-8").body(respuesta)
}

#[actix_web::main]
async fn main() -> std::io::Result<()> {
    HttpServer::new(|| App::new().service(clisoap2)).bind(("127.0.0.1", 8000))?.run().await
}